Dockerfile

```diff
@@ -50,73 +50,71 @@ RUN apt-get update -qq && \
       tzdata
 
 # Install application gems
+COPY bin ./bin
+
 COPY Gemfile Gemfile.lock ./
 RUN gem update --system && gem install bundler
 
-COPY ./bin/bundle ./bin/bundle
-RUN ./bin/bundle install && \
-    rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git
+RUN bundle config set --local path /rails/.cache/bundle
+
+# Use a cache mount to reuse old gems
+RUN --mount=type=cache,sharing=private,target=/rails/.cache/bundle \
+  mkdir -p vendor && \
+  bundle install && \
+  cp -ar /rails/.cache/bundle vendor/bundle && \
+  bundle config set path vendor/bundle
```

Comment:
...and then here again, in the `RUN` command immediately after. I don't fully understand the meaning behind this, can you explain?